### PR TITLE
Add dependency from AspNetCore targeting pack to dotnet Core targeting pack

### PR DESCRIPTION
**Pull request by pakrym**
Fixes https://github.com/aspnet/AspNetCore/issues/8284

### CODE REVIEW

File: src/Installers/Rpm/TargetingPack/Rpm.TargetingPack.rpmproj [hunk 1]
```diff
@@ -27,6 +27,7 @@
     </ProjectReference>
 
     <InstallerOwnedDirectory Include="$(RpmPackageInstallRoot)packs/$(TargetingPackName)/" />
+    <RpmDependency Include="dotnet-targeting-pack-$(AspNetCoreMajorVersion).$(AspNetCoreMinorVersion)" Version="$(MicrosoftNETCoreAppPackageVersion.Split('-')[0])" />
```
Comment (JunTaoLuo) [inline src/Installers/Rpm/TargetingPack/Rpm.TargetingPack.rpmproj]: Why is the package version different between debian and rpm installers? `DotnetRuntimeDependencyVersion` vs `MicrosoftNETCoreAppPackageVersion.Split('-')[0]`?

Comment (pakrym) [reply src/Installers/Rpm/TargetingPack/Rpm.TargetingPack.rpmproj]: Honestly, I just mirrored what was done in runtime installers but results of both operations would be the same. Also, Debian case uses token replacement and doesn't allow `Split`

Comment (JunTaoLuo) [reply src/Installers/Rpm/TargetingPack/Rpm.TargetingPack.rpmproj]: Feels strange that they are slightly different but okay.

**Review by JunTaoLuo — APPROVED** [review]

**Comment by natemcmaster** [comment]
Sorry for reviewing after you merged, but I think this code won't work for patches. This assumes the runtime version always matches the targeting pack version. The targeting pack is not expected to rev beyond 3.0.0, so this will break in 3.0.1. We should add a separate variable for the targeting pack dependency that accounts for this difference.

**Comment by JunTaoLuo** [comment]
Filed https://github.com/aspnet/AspNetCore/issues/8633.

**Pull request merged**